src/common/databaseFilters.test.ts

```diff
@@ -0,0 +1,426 @@
+import { beforeEach, expect, it } from "@jest/globals";
+import { familySearch, fullNameSearch, phoneSearch, postcodeSearch } from "./databaseFilters";
+import { DbParcelRow } from "@/databaseUtils";
+import { DbQuery, ServerSideFilterMethod } from "@/components/Tables/Filters";
+
+const logID = "a2adb0ba-873e-506b-abd1-8cd1782923c8";
+
+jest.mock("@/logger/logger", () => ({
+    logErrorReturnLogId: jest.fn(() => Promise.resolve(logID)),
+}));
+
+jest.mock("@/supabaseClient", () => {
+    return { default: jest.fn() };
+});
+
+const mockQueryWithOrAppended = jest.fn();
+const mockOr: jest.Mock = jest.fn(() => mockQueryWithOrAppended);
+
+describe("test for full name filter", () => {
+    let mockDbQuery: DbQuery<DbParcelRow>;
+    let fullNameFilter: ServerSideFilterMethod<DbParcelRow, string>;
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        // Arrange
+        mockDbQuery = {} as DbQuery<DbParcelRow>;
+        mockDbQuery.or = mockOr;
```

Comment:
Because we only define a mock function for `or`, if in future the filters change to call another operator (like `and`) at the top level, the test will error.